server/channels/app/scheduled_post_test.go

```diff
@@ -578,3 +578,86 @@ func TestUpdateScheduledPost(t *testing.T) {
 		require.Equal(t, model.ScheduledPostErrorUnknownError, createdScheduledPost.ErrorCode)
 	})
 }
+
+func TestDeleteScheduledPost(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("base case", func(t *testing.T) {
+		// first we'll create a scheduled post
+		scheduledPost := &model.ScheduledPost{
+			Draft: model.Draft{
+				CreateAt:  model.GetMillis(),
+				UserId:    th.BasicUser.Id,
+				ChannelId: th.BasicChannel.Id,
+				Message:   "this is a scheduled post",
+			},
+			ScheduledAt: model.GetMillis() + 100000, // 100 seconds in the future
+		}
+		createdScheduledPost, appErr := th.App.SaveScheduledPost(th.Context, scheduledPost)
+		require.Nil(t, appErr)
+		require.NotNil(t, createdScheduledPost)
+
+		fetchedScheduledPost, err := th.Server.Store().ScheduledPost().Get(scheduledPost.Id)
+		require.NoError(t, err) // This will produce error as the row doesn't exist
+		require.NotNil(t, fetchedScheduledPost)
+		require.Equal(t, createdScheduledPost.Id, fetchedScheduledPost.Id)
+		require.Equal(t, createdScheduledPost.Message, fetchedScheduledPost.Message)
+
+		// now we'll delete it
+		var deletedScheduledPost *model.ScheduledPost
+		deletedScheduledPost, appErr = th.App.DeleteScheduledPost(th.Context, th.BasicUser.Id, scheduledPost.Id, "connection_id")
+		require.Nil(t, appErr)
+		require.NotNil(t, deletedScheduledPost)
+
+		require.Equal(t, scheduledPost.Id, deletedScheduledPost.Id)
+		require.Equal(t, scheduledPost.Message, deletedScheduledPost.Message)
+
+		// try to fetch it again
+		reFetchedScheduledPost, err := th.Server.Store().ScheduledPost().Get(scheduledPost.Id)
+		require.Error(t, err) // This will produce error as the row doesn't exist
+		require.Nil(t, reFetchedScheduledPost)
+	})
+
+	t.Run("should not allow deleting someone else's scheduled post", func(t *testing.T) {
+		// first we'll create a scheduled post
+		scheduledPost := &model.ScheduledPost{
+			Draft: model.Draft{
+				CreateAt:  model.GetMillis(),
+				UserId:    th.BasicUser.Id,
+				ChannelId: th.BasicChannel.Id,
+				Message:   "this is a scheduled post",
+			},
+			ScheduledAt: model.GetMillis() + 100000, // 100 seconds in the future
+		}
+		createdScheduledPost, appErr := th.App.SaveScheduledPost(th.Context, scheduledPost)
+		require.Nil(t, appErr)
+		require.NotNil(t, createdScheduledPost)
+
+		fetchedScheduledPost, err := th.Server.Store().ScheduledPost().Get(scheduledPost.Id)
+		require.NoError(t, err) // This will produce error as the row doesn't exist
```

Comment:
The comment throws me off, because we are no expecting any error. Probably copy paste error from line 618?

Reply:
Removed the comment. It was copy-pasted from original test case.